src/gui/folderman.cpp

```diff
@@ -1473,11 +1477,24 @@ void FolderMan::editFileLocally(const QString &accountDisplayName, const QString
         return;
     }
 
-    // In case the VFS mode is enabled and a file is not yet hydrated, we must call QDesktopServices::openUrl from a separate thread, or, there will be a freeze.
-    // To avoid searching for a specific folder and checking if the VFS is enabled - we just always call it from a separate thread.
-    QtConcurrent::run([foundFiles] {
-        QDesktopServices::openUrl(QUrl::fromLocalFile(foundFiles.first()));
-    });
+    const auto localFilePath = foundFiles.first();
+    const auto folderForFile = folderForPath(localFilePath);
+
+    if (!folderForFile) {
+        showError(accountFound, tr("Could not find a folder to sync."), relPath);
+        return;
+    }
+    
+    const auto relPathSplit = relPath.split(QLatin1Char('/'));
+    if (relPathSplit.size() > 0) {
+        Systray::instance()->createEditFileLocallyLoadingDialog(relPathSplit.last());
+    } else {
+        showError(accountFound, tr("Could not find a file for local editing. Make sure its path is valid and it is synced locally."), relPath);
+        return;
+    }
+    folderForFile->setProperty(editFileLocallyName, localFilePath);
+    folderForFile->startSync();
+    QObject::connect(folderForFile, &Folder::syncFinished, this, &FolderMan::slotSyncFinishedBeforeOpeningForLocalEditing);
```

Comment:
this is too brittle
please use a lambda with capture for the file name
clean up in case of errors will be easier and you will also have a much easier way to track the filename once sync is completed

Reply:
@mgallien Why is it brittle? If the slot is invoked by a proper `Folder` instance, then, it will have a proper `localFilePath` set. The reason for making it a slot instead of lambda is that I could disconnect a signal, otherwise, you will end up having the file opened each time the sync is finished after any file was opened via this signal, as the connection to lambda doesn't go anywhere.

Anyways, I have implemented an alternative solution with a `QMap` of connections, even though, I'd prefer to just have a slot and make it all simpler.

Reply:
@mgallien my bad, fixed